sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -2034,14 +1956,6 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
       messageParameters = Map.empty)
   }
 
-  def fieldIndexOnRowWithoutSchemaError(): SparkUnsupportedOperationException = {
```

Comment:
Yeah, this is exactly what I want to do.

cc @MaxGekk @itholic Can we confidently clean up these unused functions?

Reply:
These functions have been moved to DataTypeErrors.
Because these are not APIs, 
Can we do this?

Reply:
Okay, after this GA runs, I'll revert it.

Reply:
Yep, I think so.